homeassistant/components/debugpy/__init__.py

```diff
@@ -0,0 +1,79 @@
+"""The Remote Python Debugger integration."""
+from asyncio import Event
+import logging
+from threading import Thread
+from typing import Optional
+
+import debugpy
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant, ServiceCall
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.service import async_register_admin_service
+from homeassistant.helpers.typing import ConfigType
+
+DOMAIN = "debugpy"
+CONF_WAIT = "wait"
+CONF_START = "start"
+SERVICE_START = "start"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Optional(CONF_HOST, default="0.0.0.0"): cv.string,
+                vol.Optional(CONF_PORT, default=5678): cv.port,
+                vol.Optional(CONF_START, default=True): cv.boolean,
+                vol.Optional(CONF_WAIT, default=False): cv.boolean,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Remote Python Debugger component."""
+    conf = config[DOMAIN]
+
+    async def debug_start(
+        call: Optional[ServiceCall] = None, *, wait: bool = True
+    ) -> None:
+        """Start the debugger."""
+        debugpy.listen((conf[CONF_HOST], conf[CONF_PORT]))
```

Comment:
Should this check to see if its already listening ?

Reply:
Yes/no?

Ideally, yes, but without starting going into checking sockets manually... As the library doesn't provide the means for it.

There is a check for seeing if a debug client is connected, however, it can still be listening if that is the case.

What will happen now, if it is called again, it will throw an exception, which is handled fine by HA in a service call. The service call would show the error.

As the service call is the only one being able to create it later, I think the current handling is fine and simple.

Reply:
Seems like its ok to let it blow up since its a debugger integration anyways.